src/script/sign.h

```diff
@@ -32,6 +32,9 @@ class BaseSignatureCreator {
     /** Create a singular (non-script) signature. */
     virtual bool CreateSig(const SigningProvider& provider, std::vector<unsigned char>& vchSig, const CKeyID& keyid, const CScript& scriptCode, SigVersion sigversion) const =0;
     virtual bool CreateSchnorrSig(const SigningProvider& provider, std::vector<unsigned char>& sig, const XOnlyPubKey& pubkey, const uint256* leaf_hash, const uint256* merkle_root, SigVersion sigversion) const =0;
+
+    /** Choose the largest stack size for worst case size estimation when using this signer */
+    virtual bool UseLargest() const =0;
```

Comment:
slightly philosophical question: If the wallet knows how to satisfy two script paths, will it use the more expensive one?

Reply:
No. This change only affects dummy signing, but for actual signing, it will still use smallest solution.